### ISSUE TITLE
Does not patch root file (App.js) in react-native project

### ISSUE THREAD

**Issue by esthor** [getsentry/sentry-wizard#184]
### Environment

React Native

How do you use Sentry?
Sentry SaaS

Which SDK and version?
4.2.3 (latest stable)
5.0.0-alpha.1 (latest alpha, for react-native 0.69+ support)

### Steps to Reproduce

1. Run `npx @sentry/wizard -i reactNative -p ios android`
2. You get a `✅ Patched App.js file.` message in the terminal
3. You're App.js isn't patched 😭 

### Expected Result
App.js to be patched.

### Actual Result

No App.js or js files were updated.

Looking at the source in `lib/Steps/Integrations/ReactNative.ts` I can see that this message will just be printed regardless of whether a successful patch occurred.

Now...here's two spots where it can get real juicy...

**Tree traversal**: It's common to have the `App.js` "root" component in the root directory of a project or in a `src` folder...it can really be anywhere.
1. one naive approach might be to traverse the whole projects file structure until you find an `App.js` and then assume that's the one to patch. You'd then need to consider priority and ignore `node_modules` of course, and it could get messy for other reasons..._[foreshadowing 👻 ]_
2. Another naive approach might be to look at the `index.js` file and see what component it passes to the `AppRegistry.registerComponent()` function call and then look where that is imported from...because that's going to be your "root" component that you want to patch. Problem here is folks might not have a `index.js` because they may be defining another entry file or heck, they might have it as a `index.jsx` or `index.ts` or `index.tsx` even if they keep named "index"...but that isn't even the spookiest... _[foreshadowing intensifies 👻 👻 ]_
3. Ok, well, actually, I've pretty much said the biggest issues with a naive approach of trying to be smart/automated about pattern matching. Even freshly init'd react native projects aren't going to follow a standard pattern due to the freedom the `react-native-template-*` feature of `react-native` cli gives template authors. You could have nested root components of any name (`main` and `root` are other common ones). So, what could you do? ...
4. So, you could make it an interactive CLI. Just ask me where my App.js or equivalent file is and what it's called. 🤷 
5. At least, don't tell me it patched a file that doesn't exist 🙏  especially after I've had too much coffee and spare time to write a novel of a GitHub issue.

**Comment by krystofwoldrich** [comment]
Hi, thank you for the info,
I've just tried it with a clean project and the `App.js` was updated correctly.

Steps I've done...
```bash
npx react-native init MyTestProjectName

yarn add @sentry/react-native   
yarn sentry-wizard -i reactNative -p ios android
```

Could you please provide repo where the Wizard failed so we can reproduce the issue?

**Comment by esthor** [comment]
Yep. The issue is it _only_ works in that scenario. But the name of the root component, where Sentry should be init'd and wrap the app, is 100% a configuration setting. There is no strict reason why it needs to be named `App.js`. That's just what the default name is in the default template.

---

A super popular template uses Typescript, [and is recommended in the official docs](https://reactnative.dev/docs/environment-setup#optional-using-a-specific-version-or-template):
`npx react-native init AwesomeTSProject --template react-native-template-typescript`

☝️ the sentry-wizard will fail here, even though the actual code that the wizard would write in to `App.js` would be totally fine in this case, but it won't write it, because the root component filename is `App.tsx` and not `App.js`.

---

In my case, the app I was installing Sentry into was a brand new one, but init'd with [Nx](https://nx.dev/), which is quickly becoming a popular monorepo toolset. You can see the [init command and the initial folder structure here](https://nx.dev/guides/react-native). The init command:
```bash
npx create-nx-workspace happynrwl \
--preset=react-native \
--appName=mobile
```

☝️ the sentry-wizard will fail here because of the more "elaborate" folder structure (and it will also default to Typescript, so what would be the `App.js` is named `App.tsx` here as well).

---

Those are two popular ways to init a brand new react native project, both with a different location of the root `App.*` component.

If it's in an already established, non-templated scaffolding, the name of the root component could theoretically be anything, although having it be named `App.js/jsx/ts/tsx` has gotta be by far the most common.

This wizard will still fail on anything not the default template's default `App.js` file name and location.

---

Maybe a solution could be something that optimistically tries what it currently does (optimistically assuming the repo is the default `react-native init` template and no one has touched anything, and they're not using TypeScript), but then falls back to something like either:
1. `We couldn't find your root component, might be named `App.js` or similar. Please copy and paste these 3 snippets into your root component`
2. Or an interactive: `Please paste the relative path to your root component. e.g. src/App.tsx`

What do you think?

**Comment by marandaneto** [comment]
I'd suggest looking for this file in the root folder with the possible extensions and if not, in the `src` folder as well.
If still not found, the wizard should log out to the user and point them to the [manual steps](https://docs.sentry.io/platforms/react-native/#initialize).
Another option is that wizard provides another param, such as `-app myCustomSrc/MyCustomApp.tsx`.

**Comment by krystofwoldrich** [comment]
The fix is part of the [`2.1.0`](https://github.com/getsentry/sentry-wizard/releases/tag/v2.1.0) release.

**Comment by upngo** [comment]
note that @sentry-wizard@2 is not compatible with @sentry/react-native@4.8.0 (latest) so make sure to run the wizard ie via `npx @sentry-wizard@2.1.0 .....` and don't install it
